support/builder/s3migrate.sh

```diff
@@ -0,0 +1,311 @@
+        fi
+    else
+        if aws s3api list-buckets | grep ${bucket_name} >/dev/null; then
+            echo "Bucket: ${bucket_name} found!"
+            echo "Using specified bucket."
+            export S3_BUCKET=${bucket_name}
+        else
+            echo "Bucket: ${bucket_name} not found!"
+            echo "Creating bucket: "${bucket_name}
+            aws s3api create-bucket --bucket ${bucket_name} --region ${AWS_REGION}
+            export S3_BUCKET=${bucket_name}
+        fi
+    fi
+}
+
+setRegion() {
+    # Sets the region used by the bucket
+    echo ""
+    echo "==========================================================="
+    echo "Please enter the region or the minio endpoint for your bucket and press [ENTER]:"
+    echo "(us-west-1 us-east-1 || http://localhost:9000 )"
+    echo "==========================================================="
+    read region_name
+    export AWS_REGION=${region_name}
+}
+
+putArtifacts() {
+    # handles the literal upload of artifacts to the configured s3 bucket
+    BODY="${1}"
+    BUCKET="${2}"
+    REGION="${3}"
+    KEY="${4}"
+
+    # upload shellout
+    if [  ${s3type} == 'minio' ]; then
+        echo ""
+        aws --endpoint-url ${REGION} s3 cp ${BODY} s3://${BUCKET}/${KEY}
+    else
+        echo ""
+        aws s3api put-object --bucket "${BUCKET}" --key "${KEY}"  --body "${BODY}" --region "${REGION}"
+    fi
+}
+
+genS3Config() {
+    aws configure
+}
+
+credSelect() {
+    echo ""
+    read -r -p "Would you like to use these credentials? [y/N] " response
+    if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+        continue
+    else
+        echo ""
+        read -r -p "Would you like to configure with custom credentials now? [y/N]" response
+        if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+            genS3Config
+        else
+            echo "Please reconfigure your AWS credentials and re-run s3migrate."
+            exit 1
+        fi
+    fi
+}
+
+welcome() {
+    echo "==========================================================="
+    echo "###########################################################"
+    echo "################ Bldr Artifact S3 Migrate #################"
+    echo "###########################################################"
+    echo "==========================================================="
+    echo "This tool will migrate all of the locally stored packages to s3/minio."
+    echo "You must have your AWS/Minio credentials configured on the system."
+    credsConfigured=false
+    if [ $s3type = "minio" ]; then
+        echo ""
+        echo "It looks like you specified a migration to minio!"
+        echo "Minio will use whatever credentials you've configured it with."
+        echo "If those credentials don't match your aws credentials file, you"
+        echo "must specify those custom credentials."
+    fi
+
+    CREDS=( "${HOME}/.aws/credentials" "${HOME}/.aws/config" "/root/.aws/credentials" "/root/.aws/config")
```

Comment:
This doesn't work if you have AWS credentials for other tooling but don't want minio creds to be the same as AWS creds.  It would be great to ask if you want to use AWS creds for minio (default to No).

Reply:
Yep, indeed I actually have a local copy that fixes this. Will push it momentarily.